src/Language/AST/FloatValueNode.php

```diff
@@ -8,6 +8,5 @@ class FloatValueNode extends Node implements ValueNode
 {
     public string $kind = NodeKind::FLOAT;
 
-    /** @var string */
-    public $value;
+    public string $value;
```

Comment:
Why the value is not a `float` in the `FloatValueNode` ?
Same for `IntValueNode`. For `BooleanValueNode` this is fine, the value is a `boolean`.

Reply:
In the case of `BooleanValueNode`, the PHP type can accurately represent all possible values. However, that is not the case for `IntValueNode` or `FloatValueNode` - the PHP types are limited in their range, whereas the GraphQL AST is not (see https://spec.graphql.org/draft/#sec-Int-Value - it can be an arbitrarily long sequence of digits, PHP `int` is limited to 32 bytes).

Scalar types are based upon those value nodes and impose certain restrictions (see https://spec.graphql.org/draft/#sec-Int), which allows us to parse incoming values of `Int` to a native PHP `int` (see https://github.com/webonyx/graphql-php/blob/7a78690e99548de8155f605772eecf615329ee16/src/Type/Definition/IntType.php#L70-L81). There may be other numeric scalar types based on the AST literal `IntValueNode` which do not fit a PHP `int`, e.g. `BigInt`.